pkg/utils/utils.go

```diff
@@ -168,3 +173,50 @@ func BytesString(bytes uint64, accuracy int) string {
 
 	return fmt.Sprintf("%s %s", stringValue, unit)
 }
+
+// WriteFileToTARArchive writes a new file with name=filename and content=contentReader to archiveWriter
+func WriteFileToTARArchive(filename string, contentReader io.Reader, archiveWriter *tar.Writer) error {
+	if filename == "" {
+		return errors.New("filename must not be empty")
+	}
+
+	if contentReader == nil {
+		return errors.New("contentReader must not be nil")
+	}
+
+	if archiveWriter == nil {
+		return errors.New("archiveWriter must not be nil")
+	}
+
+	tempfile, err := ioutil.TempFile("", "")
```

Comment:
explain that you use the tempFile for size calculation

Reply:
if you write a file to a tar archive, you must know the size upfront (gets written into the preceding file header)

Reply:
i mean a comment that the tempfile only exists for that reason